### ISSUE TITLE
Underwater digging should create water, not air

### ISSUE THREAD

**Issue by msteiger** [MovingBlocks/Terasology#1513]
Currently, the `BlockEntitySystem` replaces any block that is destroyed with an air block. 

I suggest adding a check for neighboring blocks. If water is either at one of the sides or above the destroyed block, it should be filled with water.

As a result, a single water block in the desert would lead to the generation of infinitely many water blocks below and around it. This is obviously wrong, but for the more common cases (lakes, sea) this would be correct.

Distinguishing between finite and infinite water bodies could be done in a later step (maybe based on the biome? See #1512).

**Comment by Cervator** [comment]
I put some comments in #1512 - I bet if we isolate an ocean biome you can let the water there be infinite and automatically create water when you dig down below existing ocean water. At least until we find a more thorough approach with overhauled liquid system + natural propagation.

In lakes I would question the auto-creation of more water, just from them being relatively small compared to the sea. Really need the more proper water propagation there IMHO

**Comment by skaldarnar** [comment]
I did create a new module containing a simple water propagation system which fixes this issue: 
https://github.com/Terasology/SimpleLiquids
I'll attack proper liquid propagation and finite/infinite water sources next.

The reason I did not put the code right into `BlockEntitySystem` but created a new module/system is that I don't want to be tied to any form of implementation. The module can easily be exchanged or deactivated.
